harmonica/visualization/prism.py

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) 2018 The Harmonica Developers.
+# Distributed under the terms of the BSD 3-Clause License.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# This code is part of the Fatiando a Terra project (https://www.fatiando.org)
+#
+"""
+Functions for visualizing prisms through pyvista
+"""
+import numpy as np
+
+try:
+    import pyvista
+except ImportError:
+    pyvista = None
+else:
+    import vtk
+
+
+def prisms_to_pyvista(prisms, properties=None):
+    """
+    Create a ``pyvista.UnstructuredGrid`` from a set of prisms
+
+    Parameters
+    ----------
+    prisms : list or 2d-array
+        List or 2d-array with the boundaries of the prisms.
+        Each row contains the boundaries of each prism in the following order:
+        ``west``, ``east``, ``south``, ``north``, ``bottom``, ``top``.
+    properties : dict or None (optional)
+        Dictionary with the physical properties of the prisms.
+        Each key should be a string and their value an array.
+        If None, no property will be added to the
+        :class:`pyvista.UnstructuredGrid`.
+        Default to None.
+
+    Returns
+    -------
+    pv_grid : :class:`pyvista.UnstructuredGrid`
+        :class:`pyvista.UnstructuredGrid` that represents the prisms with their
+        properties (if any).
+    """
+    # Check if pyvista are installed
+    if pyvista is None:
+        raise ValueError(
+            "Missing optional dependency 'pyvista' required for building pyvista grids."
+        )
+    # Get prisms and number of prisms
+    prisms = np.atleast_2d(prisms)
+    n_prisms = prisms.shape[0]
+    # Get the vertices of the prisms
+    vertices = np.vstack([_get_prism_vertices(prism) for prism in prisms])
```

Comment:
Hm... I wonder if there is a cost to this list comprehension for large meshes. Perhaps there is a way to vectorize this with NumPy?

Reply:
I'm sure there is place for optimization on this step. We would need to allocate this new array since our prisms are given as 6 boundaries and the vertices need the 3 coordinates of each one of the 8 vertices. But might be a way to create this array in a vectorized way without list comprehensions.

Let me experiment with it and give it a try.

Reply:
I've just pushed a more efficient way to generate the vertices using Numpy. I only iterate with Python for loops along the 6 boundaries (west, east, south, north, bottom, top) and on the 4 vertices that each one of the appear, leading to only 24 inefficient iterations, regardless of the size of the set of prisms.